src/Angor/Client/Pages/Create.razor

```diff
@@ -1172,30 +1174,50 @@
     {
         var selected = e.Value?.ToString();
 
-        if (selected != null)
+        if (selected != null && int.TryParse(selected, out int res) && res <= feeData.FeeEstimations.Fees.Count)
         {
-            if (int.TryParse(selected, out int res))
-            {
-                if (res <= feeData.FeeEstimations.Fees.Count)
-                {
-                    if (!passwordComponent.HasPassword())
-                    {
-                        showCreateModal = false;
-                        notificationComponent.ShowErrorMessage("Wallet password expired");
-                        return;
-                    }
+            feeData.FeePosition = res;
 
-                    var words = await passwordComponent.GetWalletAsync();
-                    feeData.SelectedFeeEstimation = feeData.FeeEstimations.Fees.OrderBy(fee => fee.Confirmations).ToList()[res - 1];
+            var selectedFeeEstimation = feeData.FeeEstimations.Fees.OrderBy(fee => fee.Confirmations).ToList()[res - 1];
 
-                    var accountInfo = storage.GetAccountInfo(network.Name);
-                    var unconfirmedInfo = _cacheStorage.GetUnconfirmedInboundFunds();
+            await RecalculateFee(selectedFeeEstimation.FeeRate, selectedFeeEstimation.Confirmations);
+        }
+    }
 
-                    signedTransaction = _WalletOperations.AddInputsAndSignTransaction(accountInfo.GetNextChangeReceiveAddress(), unsignedTransaction, words, accountInfo, feeData.SelectedFeeEstimation.FeeRate);
+    private async Task RecalculateFee(long customFeeRate, int confirmations = 0)
```

Comment:
Milad I realize we can simplify this much more, instead of two methods we will make one method that will take a class that has a Position or a customFee (and maybe a bool that says useCustomFee.

Then we don't need to have two methods instead we just have one method called `RecalculateFee`
And we also dont need two events on the ui component.
this simplifies the code.

you can even call that recalcualtefee method form the send method and save the duplicate code.